### PR TITLE
WR324448: minimum files for plugin

### PR DESCRIPTION
**Pull request by andrewmadden**

### CODE REVIEW

File: rule.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+    /**
+     * quizaccess_seb constructor.
+     */
+    public function __construct($quizobj, $timenow) {
+        parent::__construct($quizobj, $timenow);
+    }
+
+    /**
+     * @param quiz $quizobj
```
Comment (dmitriim) [inline rule.php]: Missing doc?

File: rule.php [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+    /**
+     * Return the SQL needed to load all the settings from all the access plugins in one DB query.
+     *
+     * Array to return contains three elements - [fields, joins, params].
+     *
+     * @param int $quizid
+     * @return array
+     */
+    public static function get_settings_sql($quizid) {
```
Comment (dmitriim) [inline rule.php]: I think the general rule of thumb (not s standard yet) to cast return type for the functions

File: version.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2019 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+$plugin->version = 2019120400;
+$plugin->release = 2019052000; // Match to version.
+$plugin->requires = 2017111300; // Moodle 3.7.
```
Comment (dmitriim) [inline version.php]: should it be 3.9 potentially?

Comment (andrewmadden) [reply version.php]: I did it off the PHP version required. I'm using some 7.1 features (class constants). Just saw I copy pasted it incorrectly. Requires should be 2019052000. 

I thought it makes sense to allow people to install it as a plugin in 3.7 and 3.8 if they want to?

Comment (dmitriim) [reply version.php]: Yes if we won't implement new api changes in core to support this plugin.

File: rule.php (hunk 1 above)
Comment (andrewmadden) [reply rule.php]: Fixed

File: rule.php (hunk 2 above)
Comment (andrewmadden) [reply rule.php]: Ah thanks, assuming you mean by casting  `function() : array {}`, I've added the return time to function sig where possible. All of these methods are stubbed from the base class.

File: version.php (hunk 3 above)
Comment (andrewmadden) [reply version.php]: Very good point. I'll try and remember to raise the version if we need to change core to integrate the plugin.

File: db/install.xml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (dmitriim) [inline db/install.xml:1]: I think PREVIOUS="id" NEXT="allowedkeys" is not required anymore

File: rule.php (hunk 1 above)
Comment (dmitriim) [reply rule.php]: As we are aiming to push this code to core, let's start to follow strict standards from very beginning. We need to have description for all parameters in all methods.

**Pull request merged**